tools/uf2conv.py

```diff
@@ -248,21 +265,12 @@ def get_drives():
             if len(words) >= 3 and words[1] == "2" and words[2] == "FAT":
                 drives.append(words[0])
     else:
-        rootpath = "/run/media"
-        if not os.path.isdir(rootpath):
-            rootpath = "/media"
-        if not os.path.isdir(rootpath):
-            rootpath = "/opt/media"
         if sys.platform == "darwin":
-            rootpath = "/Volumes"
+            possibly_anydir("/Volumes", drives)
         elif sys.platform == "linux":
-            tmp = rootpath + "/" + os.environ["USER"]
-            if os.path.isdir(tmp):
-                rootpath = tmp
-        for d in os.listdir(rootpath):
-            drives.append(os.path.join(rootpath, d))
-
-        if (len(drives) == 0) and (sys.platform == "linux"):
+            # Do first since a race condition is possible.
+            time.sleep(2)
```

Comment:
Random delay/sleeps seem to be a code smell.

The `main` code in line 386 already calls this repeatedly for up to 10s after the reset is sent, so there's no sense in sleeping here.

Reply:
The problem is that the daemon might be already trying to to this. What happens is that if it is already in-progress, it causes the daemon and the manual request seems to abort and "eject" the media. Unfortunately there's no way to tell if this is attempting to mount. Perhaps it would be best to, instead, check for 10 seconds as it does currently, and if that is taking too long, then attempt the mount hack and check again?